Test/dafny0/AndOrAmbiguity.dfy

```diff
@@ -6,56 +6,56 @@
 // The tests below check that these ambiguities result in understandable error message.
 
 method M0() returns (A: bool, B: bool)
-  ensures A && B ||  // error: and-or ambiguity
+  ensures A && B || A // error: and-or ambiguity
```

Comment:
Why is this "A" needed to have the same error message bas before? What happens if you remove the A?

Reply:
Without the 'A' one gets a cascade of errors because of the syntax error, so it is cleaner to separate the semantic error (needing parentheses) from the syntactic error (failure to parse an expression)